src/python/tests/test_statistics.py

```diff
@@ -0,0 +1,143 @@
+
+__author__ = "Ori Shadmon"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+_STATISTICS_TBL = sa.Table(
+    'statistics',
+    sa.MetaData(),
+    sa.Column('key', sa.types.CHAR(10)),
+    sa.Column('description', sa.types.VARCHAR(255)),
+    sa.Column('value', sa.types.BIGINT),
+    sa.Column('previous_value', sa.types.BIGINT),
+    sa.Column('ts', sa.types.TIMESTAMP)
+)
+
+_CONNECTION_STRING = "dbname='foglamp'"
+_KEYS = []
+
+pytestmark = pytest.mark.asyncio
+
+async def set_in_keys():
+    """Set statistics.keys column into a list to be used by test cases"""
+    stmt = sa.select([_STATISTICS_TBL.c.key])
+    try:
+        async with aiopg.sa.create_engine(_CONNECTION_STRING) as engine:
+            async with engine.acquire() as conn:
+                async for result in conn.execute(stmt):
+                    _KEYS.append(result[0].replace(" ", ""))
+    except Exception:
+        raise
+
+@pytest.allure.feature("unit")
+@pytest.allure.story("statistics")
+class TestStatistics:
+    """
+    Test the different components of src/python/foglamp/statistics_history.py
+    """
+
+    def setup_method(self):
+        """
+        Set up each test with fresh data, and _KEYS dictionary with
+        values from statistics.key column
+        """
+        _KEYS.clear()
+        os.system("psql < `locate foglamp_ddl.sql | grep 'FogLAMP/src/sql'` > /dev/null 2>&1")
+        os.system("psql < `locate foglamp_init_data.sql | grep 'FogLAMP/src/sql'` > /dev/null 2>&1")
+        event_loop = asyncio.get_event_loop()
+        event_loop.run_until_complete(set_in_keys())
+
+    def teardown_method(self):
+        """Set up each test with fresh data, and empty _KEYS dictionary"""
+        _KEYS.clear()
+        os.system("psql < `locate foglamp_ddl.sql | grep 'FogLAMP/src/sql'` > /dev/null 2>&1")
+        os.system("psql < `locate foglamp_init_data.sql | grep 'FogLAMP/src/sql'` > /dev/null 2>&1")
+
+    async def test_update_value(self):
+        """
+        Test that statistics table gets updated
+        :assert:
+            1. value gets update with new rand_value
+            2. previous value does not change
+        """
+        for key in _KEYS:
+            rand_value = random.randint(1, 10)
+            await update_statistics_value(statistics_key=key, value_increment=rand_value)
+            stmt = sa.select([_STATISTICS_TBL.c.value]).select_from(_STATISTICS_TBL).where(
+                _STATISTICS_TBL.c.key == key)
+            try:
+                async with aiopg.sa.create_engine(_CONNECTION_STRING) as engine:
+                    async with engine.acquire() as conn:
+                        async for result in conn.execute(stmt):
+                            assert result[0] == rand_value
+            except Exception:
+                raise
+
+            stmt = sa.select([_STATISTICS_TBL.c.previous_value]).select_from(_STATISTICS_TBL).where(
```

Comment:
do we need a new call? can we fetch this c.previous_value in previous select itself? and compare?